### ISSUE TITLE
how to handle RefreshRemoteApplicationEvent in config client using default profile?

### ISSUE THREAD

**Issue by yidongnan** [spring-cloud/spring-cloud-config#479]
When I use config-monitor,if i change config file.I can see config server push a event.
in **/trace**

```
{
timestamp: 1471486891350,
info: {
signal: "spring.cloud.bus.sent",
type: "RefreshRemoteApplicationEvent",
id: "0c323a59-4824-4054-ab0f-964198695bf7",
origin: "config-server:native:8100",
destination: "service-a-default:**"
}
},
{
timestamp: 1471486890743,
info: {
signal: "spring.cloud.bus.sent",
type: "RefreshRemoteApplicationEvent",
id: "14ec9042-2337-43bb-8939-4b67038b6c6c",
origin: "config-server:native:8100",
destination: "service-a:default"
}
},
{
timestamp: 1471486890073,
info: {
signal: "spring.cloud.bus.sent",
type: "RefreshRemoteApplicationEvent",
id: "83f0ba67-2dd8-45b9-9d55-0f1a26a5b4b3",
origin: "config-server:native:8100",
destination: "service:a-default"
}
},
```

In my config client(service-a) **/trace**

```
{
timestamp: 1471486891350,
info: {
signal: "spring.cloud.bus.sent",
type: "RefreshRemoteApplicationEvent",
id: "0c323a59-4824-4054-ab0f-964198695bf7",
origin: "config-server:native:8100",
destination: "service-a-default:**"
}
},
{
timestamp: 1471486890741,
info: {
signal: "spring.cloud.bus.sent",
type: "RefreshRemoteApplicationEvent",
id: "14ec9042-2337-43bb-8939-4b67038b6c6c",
origin: "config-server:native:8100",
destination: "service-a:default"
}
},
{
timestamp: 1471486890051,
info: {
signal: "spring.cloud.bus.sent",
type: "RefreshRemoteApplicationEvent",
id: "83f0ba67-2dd8-45b9-9d55-0f1a26a5b4b3",
origin: "config-server:native:8100",
destination: "service:a-default"
}
},
```

It mean config client receive this message.
But it not run **refresh**
So the config not change in config client
When I run **localhost:8080/refresh**.It can refresh config.

And then when I use 

```
http --form POST :8100/monitor path=service-a
```

It can refresh config.
config client(service-a) receive the event in "/trace" 

```
{
timestamp: 1471498671136,
info: {
signal: "spring.cloud.bus.sent",
type: "RefreshRemoteApplicationEvent",
id: "9009a8ba-ed29-4d0c-adea-637a7ef6f7cc",
origin: "config-server:native:8100",
destination: "service-a:**"
}
},
{
timestamp: 1471498671136,
info: {
signal: "spring.cloud.bus.ack",
event: "RefreshRemoteApplicationEvent",
id: "9009a8ba-ed29-4d0c-adea-637a7ef6f7cc",
origin: "service-a:8080",
destination: "service-a:**"
}
},
{
timestamp: 1471498669456,
info: {
signal: "spring.cloud.bus.sent",
type: "RefreshRemoteApplicationEvent",
id: "afada5bc-e057-466f-9ec5-4ac5dbd5737c",
origin: "config-server:native:8100",
destination: "service:a"
}
},
```

**Comment by ryanjbaxter** [comment]
Based on your description I am not sure I understand what the problem is, can you be more clear?

**Comment by yidongnan** [comment]
@ryanjbaxter     
when I change config in config server.
config client received push notification.
but it not refresh config.

**Comment by dsyer** [comment]
The problem is partly that "default" is a profile but it isn't included in the service instance id for the bus (because it's the default).  So you made a change to a file called `service-a-default.yml` (or similar for a properties file) and the monitor is trying to guess the service id and profile.  You can just move your config file to `service-a.yml` and it will work as expected I think.

**Comment by yidongnan** [comment]
@dsyer It mean it work if profile is not "default"?

**Comment by dsyer** [comment]
I mean it should work even if the profile is default, but not if the filename has "-default" in it.

**Comment by yidongnan** [comment]
@dsyer When I rename to service-a.yml, it work in "default" profile. Thks.
But in "docker" profile, my config name is service-a-docker.yml, It work not normally.
Code links => https://github.com/yidongnan/spring-cloud-netflix-example/tree/dev

**Comment by dsyer** [comment]
The logs look like maybe you are pinging the /health endpoint every 10 seconds. I didn't see any remote refresh events, so it doesn't seem relevant to this discussion.

**Comment by yidongnan** [comment]
@dsyer  the logs is pinging the /health endpoint...thks. :smile:
:question:  service-a running in "docker" profile. config-server running in "docker,native" profile
when I change service-a-docker.yml in config server. service-a can't refresh config
when I change service-a.yml in config server. service-a  can receive event and refresh config

**Comment by dsyer** [comment]
I'm not 100% sure I follow that yet. The config server can run in whatever profile it likes, and it shouldn't affect the behaviour of the client apps. I can see there might be a problem with the matching algorithm in the bus for multiple profiles (in the client app), but that doesn't seem to be your use case.

**Comment by yidongnan** [comment]
@dsyer  I think I find out.

https://github.com/spring-cloud/spring-cloud-bus/blob/v1.1.0.RELEASE/spring-cloud-bus/src/main/java/org/springframework/cloud/bus/ServiceMatcher.java#L33

in "default" profile. event destinationService is service-a:** ,it match service-a:docker:8080
in "docker" profile. event destinationService is service-a:docker ,it don't match service-a:docker:8080 

It fix in #472 

But if service-a active  multiple profile. It don't match.
matcher.match("service-a:docker:**", "service-a:docker,production:8080")  // false
